db_test.go

```diff
@@ -15,6 +16,28 @@ type testUser struct {
 	ID       string
 }
 
+type testStatefulUser struct {
+	Username  string      `type:"string" unique:"Username"`
```

Comment:
can be better with `reflect.TypeOf()` instead of Struct Tag

Such as.
```go
    test1 := "test string"
    test2 := 30
    test3 := 1.5

    fmt.Println(reflect.TypeOf(test1)) // string
    fmt.Println(reflect.TypeOf(test2)) // int
    fmt.Println(reflect.TypeOf(test3)) // float64
```

Reply:
Yes, of course, I thought about it before adding support for `INDEX, UNIQUE, ASSERT, VALUE, SCHEMAFULL, SCHEMALESS`.
It is not about types only, they are many needed descriptive options.

Reply:
> It is not about types only, they are many needed descriptive options.

I agree but we can use variable type as default type then if there was a type struct tag that can override the type.
Also like you said struct tags are needed but for for simpler use cases, this will be more convenient to use for end-user.

Something like this
```go
type testUser struct {
	Username string // No tag so its type will be string 
	Something int64  `type:"int"` // this will override
	SomeBool   bool  `index:"SomeBool,Something"` // They can use index and descriptive options with struct tags.
}
```

Reply:
By default, if no type is provided then it will be converted to the closest possible type.